docs/guide/data_types/timeseries.rst

```diff
@@ -441,30 +306,103 @@ console view: ::
     2011-06-07T23:59:53.538999000   1e-09 1.5985e-07
     2011-06-07T23:59:57.631999000   1e-09 1.5985e-07
 
-and the more sophisticated browser view using the `~astropy.table.Table.show_in_browser`
-method: ::
+and the more sophisticated browser view using the `~astropy.table.Table.show_in_browser` method: ::
 
     >>> table.show_in_browser(jsviewer=True)  # doctest: +SKIP
 
-For further details about editing Astropy tables you can read the `astropy
-documentation website <https://docs.astropy.org/en/stable/table/>`_.
+For further details about editing Astropy tables you can read the `Astropy documentation website <https://docs.astropy.org/en/stable/table/>`_.
+
+.. _custom-timeseries:
+
+5. Creating Custom TimeSeries
+=============================
+
+Sometimes you will have data that you want to create into a TimeSeries.
+You can use the factory to create a `~sunpy.timeseries.GenericTimeSeries` from a variety of data sources currently including `pandas.DataFrame` and `astropy.table.Table`.
+
+5.1 Creating a TimeSeries from a Pandas DataFrame
+-------------------------------------------------
+
+A TimeSeries object must be supplied with some data when it is created.
+The data can either be in your current Python session, in a local file, or in a remote file.
+Let's create some data and pass it into a TimeSeries object: ::
+
+    >>> import numpy as np
+    >>> intensity = np.sin(np.arange(0, 12 * np.pi, ((12 * np.pi) / (24*60))))
+
+The first line imports the numpy module used to create and store the data.
+The second line creates a basic numpy array of values representing a sine wave.
+We can use this array along with a suitable time storing object (such as Astropy `~astropy.time` or a list of `datetime` objects) to make a Pandas `~pandas.DataFrame`.
+A suitable list of times must contain the same number of values as the data, this can be created using: ::
+
+    >>> import datetime
+    >>> base = datetime.datetime.today()
+    >>> times = [base - datetime.timedelta(minutes=x) for x in range(24*60, 0, -1)]
+
+The Pandas `~pandas.DataFrame` will use the dates list as the index: ::
+
+    >>> from pandas import DataFrame
+    >>> data = DataFrame(intensity, index=times, columns=['intensity'])
+
+This `~pandas.DataFrame` can then be used to construct a TimeSeries: ::
+
+    >>> import sunpy.timeseries as ts
+    >>> ts_custom = ts.TimeSeries(data)
+
+Furthermore we could specify the metadata/header and units of this time series by sending them as arguments to the factory: ::
+
+    >>> from collections import OrderedDict
```

Comment:
I wonder if we can remove this, we shouldn't need to use an OrderedDict from my understanding.